pynicotine/search.py

```diff
@@ -47,6 +48,44 @@ def __init__(self, token=None, term=None, mode="global", room=None, users=None,
         self.users = users
         self.is_ignored = is_ignored
 
+        self.max_results = config.sections["searches"]["max_displayed_results"]
+        self.num_results = 0
+        self.results = {}
+        self.term_words_include = []
+        self.term_words_ignore = []
+
+        for word in term.lower().split():
+            if word.startswith("*"):
+                if len(word) > 1:
+                    self.term_words_include.append(word[1:])
+
+            elif word.startswith("-"):
+                if len(word) > 1:
+                    self.term_words_ignore.append(word[1:])
+
+            else:
+                self.term_words_include.append(word)
+
+
+class SearchResult:
+
+    __slots__ = ("num", "user", "file_path", "size", "attributes", "speed", "country_code", "queue",
+                 "has_free_slots", "is_private")
+
+    def __init__(self, num=None, user=None, file_path=None, size=None, attributes=None, speed=None, country_code=None,
+                 queue=None, has_free_slots=None, is_private=False):
+
+        self.num = num
+        self.user = user
+        self.file_path = file_path
```

Comment:
`file_path` -> `fullpath` or `full_path`

Also consider adding `folder` and `filename` here would help to reduce the amount of split() and join() operations elsewhere.